paper/outline.md

```diff
@@ -34,66 +34,74 @@
-        - Template for creating HyP3 containerized plugins
-            - Defines the entrypoint so that HyP3 can understand how to run the container
-        - AWS ECR repositories straightforward to create for cookiecutter containers
-    - HyP3 Access
+        - In 2022 HyP3 was migrated to NASA Earthdata Cloud with stable funding from NASA
+    - HyP3 Architecture
+        - platform 
+            - Figure: [Architecture diagram](https://drive.google.com/file/d/1wZUPGl4pY1qF5ojNSODJ6mF2CAiM-EaT/view?usp=sharing)
+                - (buzzwords: - API Gateway, Lambda, DynamoDB, Step Functions, Batch, S3, Docker/Container, ECR, CloudWatch, CloudFormation)
+            - Cloud-native (AWS) API-based processing workflow for satellite imagery 
+            - High throughput, scalable and cost-effective computing, not high performance computing
+            - Designed to work with any image processing pipeline via use of containerized batch jobs for processing
+            - Infrastructure as code with AWS CloudFormation Templates
+                - CF templates make new, specialized deployments easy
+            - API Gateway receives GET, POST requests
+            - Post Request logged in DynamoDB NoSQL database
+            - AWS Lambda runs on a schedule and watches DynamoDB jobs table for new (PENDING) jobs
+            - AWS Step Functions defines workflow for a particular job type
+                - Delegate appropriate plugin container and memory requirements for job type
+                - Watch for failures, log, and automatically retry
+                - Upload completed jobs and updates DynamoDB table with job status (FAILED, SUCCEEDED)
+            - AWS Batch runs appropriate plugin container for job type
+                - UTILIZES SPOT INSTANCES FOR COST MANAGEMENT
+                - Scales according to usage compared to monthly budget allowance (using AWS Cost Explorer)
+            - AWS S3 buckets house results download 
+                - Output data are publicly available for 2 weeks
+        - plugins
+            - production (generally available in vertex)
+                - HyP3 GAMMA
+                    - RTC and InSAR processing workflows on demand
+                    - Commercial software 
+                - HyP3 autoRIFT
+                    - Specialized algorithm finding pixel displacement between two radar or optical images developed by the ITS_LIVE team at JPL `[@Lei:2021]`
+            - custom (available in select custom deployments)
+                - DockerizedTopsApp
+                    - open source InSAR processing for Sentinel-1 Geocoded Unwrapped Interferograms (GUNW)
+                - Watermap (ASF Tools?)
+                    - Processing utilizing RTC products with different polarizations to estimate surface water extent and depth
+                - Riverwidth
+                    - processing utilizing RTC product with different polarization to estimate surface water extent and river widths  
```

Comment:
I think it makes the most sense to talk about HyP3 as comprised of:
* the platform
* the plugins

So I'd like the Architecture section to have a sub-section for both. 

I'm not sure if we should only talk about the production (generally available) plugins here and move all the custom ones to a section about custom deployments, or just group them all together. That might fall out of the large writing effort.

Reply:
That makes sense to me. In that vein, do you think the HyP3 Access section would make sense before talking about the API Gateway or is it best at the end?

Reply:
I'm not sure. I suspect when writing it'll most naturally flow like:
* what it is
* why it is needed
* how to use it
* how we built it

But I the last two could go in the opposite order.  I think we'll find out as we write and right now the outline gets us moving in that direction so I'm happy with it :-D